### ISSUE TITLE
Any Good Documentation?

### ISSUE THREAD

**Issue by rebootcode** [silgy/silgy#3]
Documentation is very random: 

Is there any documentation that has one by one example from "Opening Base URL to Session to Database and Logout etc.,?"

It's very complicated to understand the documentation

**Comment by silgy** [comment]
You're right, there should be a complete example. I've been planning this since the beginning (there are even links to phantom Toy Facebook scattered through [Functions and macros](https://github.com/silgy/silgy/wiki/Silgy-functions-and-macros)) and you just poked me to finally do it.

**Comment by rebootcode** [comment]
This is so critical - functions and macros only help to understand what is function all about - Nothing more than that.

Waiting for good documentation. Thank you so much for such framework. Do you have any future plan as well to grow more? or do you want to keep it simple as it is?

**Comment by silgy** [comment]
Yes, definitely there are plans, more than my spare time. I wouldn't like to bloat it though, I've got the whole philosophy around this and I'm planning on starting a blog -- there will be space to share what's going on and why. Thanks!

**Comment by rebootcode** [comment]
Alright! Can't wait to see it.

Thank you

**Comment by sukhchain9935** [comment]
Yes, a proper documentation would be nice.

**Comment by silgy** [comment]
Hi,

I agree, documentation is needed and it's on the way. Realistically, I'll be able to catch up around the end of the year.

As for the client application, the basic example would be lib/silgy_watcher.c. You could skip parameters reading from the conf file (if you don't want that) but you have to ensure that your program contains:

* include silgy.h
* silgy_lib_init
* CALL_REST / CALL_HTTP

Link it with -D SILGY_WATCHER switch and silgy_lib.c (see the last part in src/m script). SILGY_WATCHER will exclude server part from the library.

Let me know if that works, please, I've got several client programs working (i.e. eBay API client for retrieving listings).
